### PR TITLE
Refactored EPP resources creation and added tests

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Refactored EPP resources creation and added tests

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/24

### CODE REVIEW

File: test/utils/utils.go [hunk 1]
```diff
@@ -66,7 +76,111 @@ func NewEventingCR(opts ...EventingOption) *v1alpha1.Eventing {
...
+			Verbs:     []string{"get", "list", "watch"},
+		},
+	}
+	return reflect.DeepEqual(wantRules, clusterRole.Rules)
+}
+
+func IsEPPClusterRoleBindingCorrect(clusterRoleBinding rbacv1.ClusterRoleBinding, eventingCR v1alpha1.Eventing) bool {
+	wantRoleRef := rbacv1.RoleRef{
+		Kind:     "ClusterRole",
+		Name:     fmt.Sprintf("%s-%s", eventingCR.GetName(), PublisherProxySuffix),
```
Comment (muralov) [inline test/utils/utils.go:171]: nitpick: pkg/eventing/utils.go`GetEPPClusterRoleName` can be used here.

File: test/utils/utils.go [hunk 2]
```diff
@@ -66,7 +76,111 @@ func NewEventingCR(opts ...EventingOption) *v1alpha1.Eventing {
...
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+	wantSubjects := []rbacv1.Subject{
+		{
+			Kind:      "ServiceAccount",
+			Name:      fmt.Sprintf("%s-%s", eventingCR.GetName(), PublisherProxySuffix),
```
Comment (muralov) [inline test/utils/utils.go:177]: nitpick: pkg/eventing/utils.go`GetEPPServiceAccountName` can be used here.

File: pkg/eventing/manager.go [hunk 3]
```diff
@@ -84,6 +83,8 @@ func (em *EventingManager) applyPublisherProxyDeployment(
 	eventing *v1alpha1.Eventing,
 	backendType v1alpha1.BackendType) (*appsv1.Deployment, error) {
 	var desiredPublisher *appsv1.Deployment
+	// update service account name.
+	em.backendConfig.PublisherConfig.ServiceAccount = GetEPPServiceAccountName(*eventing)
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply pkg/eventing/manager.go:87]: https://github.com/kyma-project/eventing-manager/pull/29#discussion_r1280222136

File: pkg/eventing/manager.go
[the remark this answers is not in the document]
Comment (muralov) [reply pkg/eventing/manager.go:1]: you can take the decision as it is nitpick.

File: pkg/eventing/manager_test.go [hunk 4]
```diff
@@ -509,3 +513,137 @@ func Test_ConvertECBackendType(t *testing.T) {
...
+
+func Test_DeployPublisherProxyResources(t *testing.T) {
+	t.Parallel()
+
+	// given
+	newScheme := runtime.NewScheme()
+	require.NoError(t, v1alpha1.AddToScheme(newScheme))
+
+	// EPP deployment
+	sampleEPPDeployment := &appsv1.Deployment{
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply pkg/eventing/manager_test.go]: done

File: pkg/eventing/manager_test.go [hunk 5]
```diff
@@ -509,3 +513,137 @@ func Test_ConvertECBackendType(t *testing.T) {
...
+	findService := func(name string, objects []client.Object) (client.Object, error) {
+		for _, obj := range objects {
+			if obj.GetObjectKind().GroupVersionKind().Kind == "Service" &&
+				obj.GetName() == name {
+				return obj, nil
+			}
+		}
+
+		return nil, errors.New("not found")
+	}
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply pkg/eventing/manager_test.go]: moved to test-utils.

File: pkg/eventing/manager.go
[the remark this answers is not in the document]
Comment (mfaizanse) [reply pkg/eventing/manager.go:1]: lets keep it for now. Later we can make the naming consistent.

**Review by muralov — APPROVED** [review]

**Pull request merged**